src/msg/async/rdma/RDMAServerSocketImpl.cc

```diff
@@ -26,10 +26,9 @@ int RDMAServerSocketImpl::listen(entity_addr_t &sa, const SocketOptions &opt)
   int rc = 0;
   server_setup_socket = net.create_socket(sa.get_family(), true);
   if (server_setup_socket < 0) {
-    rc = -errno;
     lderr(cct) << __func__ << " failed to create server socket: "
                << cpp_strerror(errno) << dendl;
-    return rc;
+    return -errno;

```

Comment:
actually you should change cpp_strerror to use rc instead.  the lderr op might make a syscall and reset errno.